tests/core/data_storage/test_in_memory_storage.py

```diff
@@ -0,0 +1,113 @@
+import functools
+
+import numpy
+
+from openfisca_core import data_storage
+from openfisca_core import periods
+
+import pytest
+
+
+@pytest.fixture
+def storage():
+    return data_storage.InMemoryStorage
+
+
+@pytest.fixture
+def eternal_storage(storage):
+    return functools.partial(storage, is_eternal = True)
+
+
+@pytest.fixture
+def period():
+    return periods.period("2020")
+
+
+@pytest.fixture
+def value():
+    return numpy.array([1])
+
+
+def test___init__(storage):
+    result = storage()
+
+    assert result
+
+
+def test___init__when_is_eternal(eternal_storage):
+    result = eternal_storage()
+
+    assert result.is_eternal
+
+
+def test_put(storage, period, value):
+    storage = storage()
+    storage.put(value, period)
+
+    result = storage.get(period)
+
+    assert result == value
+
+
+def test_put_when_is_eternal(eternal_storage, value):
+    storage = eternal_storage()
+    storage.put(value, "foo")
+
+    result = storage.get("bar")
```

Comment:
This doesn't really make sense. The argument of `get` is a period, not an arbitrary string. This test commits the implementation to happily support arbitrary strings, which isn't the intent; the intent is to return the same value for all periods.

Reply:
Hello, I'm just trying to map the current behaviour, in an effort of exposing redundancies. Why? Because those are IMHO low-hanging fruits that'll helps us do what we want to do after: get rid of the current implementation.

Reply:
Changed !